middleware/protect.js

```diff
@@ -17,22 +17,22 @@
 
 const UUID = require('./../uuid');
 
-function forceLogin (keycloak, request, response) {
-  let host = request.hostname;
+function forceLogin (keycloak, request, response, next) {
   let headerHost = request.headers.host.split(':');
+  let host = headerHost[0];
   let port = headerHost[1] || '';
-  let protocol = request.protocol;
+  let protocol = request.headers['x-forwarded-proto'] || (request.isSecure() ? 'https' : 'http');
   let hasQuery = ~(request.originalUrl || request.url).indexOf('?');
 
-  let redirectUrl = protocol + '://' + host + (port === '' ? '' : ':' + port) + (request.originalUrl || request.url) + (hasQuery ? '&' : '?') + 'auth_callback=1';
+  let redirectUrl = protocol + '://' + host + (port === '' ? '' : ':' + port) + '/saml_callback' + (hasQuery ? '&' : '?') + 'auth_callback=1';
```

Comment:
Why did you need to change these paths to `/saml_callback` in 2 places?

Reply:
I figured that when the user logs out, we should just send them back to the Keycloak login screen, instead of the Outlier login page, to skip a step. So sending them to `saml_callback` does that since it'll go through the `protect` middleware which will redirect them. Basically, I'm using this `saml_callback` endpoint for both logout and login because I registered the `protect` middleware there and it does the work of determining whether the user can continue to the app or needs to log in.

Reply:
Makes a ton of sense. Is this something we can add easily as a configuration option like `logoutRedirectPath` or `logoutRedirectUrl`?

Reply:
Added something called `redirectPath` since it was common to both logout and the protect.